source/developers-guide/plugin-configuration/index.md

```diff
@@ -0,0 +1,685 @@
+            )
+        );
+}
+```
+
+<img src="img/remotecombo.png" alt="Remotecombobox" class="image-border" />
+
+### Textfield
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('text', 'text', 
+        array(
+            'label' => 'Text', 
+            'value' => NULL
+        )
+    );
+}
+```
+
+<img src="img/textfield.png" alt="Textfield" class="image-border" />
+
+### Textarea
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('textarea', 'textarea',
+        array(
+            'label' => 'Text-Area', 
+            'value' => NULL
+        )
+    );
+}
+```
+
+<img src="img/textarea.png" alt="Textarea" class="image-border" />
+
+### Timefield
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('time', 'time', 
+        array(
+            'label' => 'Time', 
+            'value' => NULL
+        )
+    );
+}
+```
+
+<img src="img/timefield.png" alt="Timefield" class="image-border" />
+
+## Configure basic settings
+Since Shopware 4 the plugin configurations of the various plugins are automatically 
+reachable via the basic settings under the "Additional settings" entry:
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('text', 'text',
+        array(
+            'label' => 'Text',
+            'value' => 'Vorauswahl',
+            'scope' => Shopware\Models\Config\Element::SCOPE_SHOP,
+            'description' => 'Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut',
+            'required' => true
+        )
+    );
+}
+```
+
+<img src="img/configuration.png" alt="Plugin configuration in base settigns" class="image-border" />
+
+If you think they are better off under another entry, such as with payment interfaces, the entry "Interface", 
+the plugin configuration can be classified as follows:
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('text', 'text',
+        array(
+            'label' => 'Text',
+            'value' => 'Vorauswahl',
+            'scope' => Shopware\Models\Config\Element::SCOPE_SHOP,
+            'description' => 'Lorem ipsum dolor sit amet, consetetur sadipscing elitr.',
+            'required' => true
+        )
+    );
+ 
+    $repository = Shopware()->Models()->getRepository('Shopware\Models\Config\Form');
+    $form->setParent(
+        $repository->findOneBy(array('name' => 'Interface'))
+    );
+}
+```
+
+<img src="img/configuration_2.png" alt="Plugin configuration in base settigns other position" class="image-border" />
+
+## Translation of configurations
+Since Shopware 4 it is possible to translate the Shopware Backend. To translate the custom plugin configurations you can use following code:
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
```

Comment:
there is also the possibility to create translations for select options

\Shopware_Plugins_Frontend_SwagPromotion_Bootstrap::createPluginConfig()